docs/advanced-features/athena.md

```diff
@@ -1,91 +1,334 @@
+3. Create an Athena Database `quilt-metadata` for that Workgroup
+
+Later we will explicitly grant Quilt access to that Bucket.
+<!--pytest-codeblocks:cont-->
+
+
+```python
+COMPANY = "mycompany"
+ATHENA_BUCKET = f"{COMPANY}-quilt-athena-output"
+ATHENA_DB = "quilt_metadata"
+ATHENA_URL = "s3://" + ATHENA_BUCKET
+ATHENA_WORKGROUP = "QuiltQueries"
+
+ARN_PREFIX = "arn:aws:s3:::"
+ARN_ATHENA = ARN_PREFIX + ATHENA_BUCKET
+
+# Create bucket in default region
+
+bucket = (
+    S3.create_bucket(Bucket=ATHENA_BUCKET)
+    if REGION == "us-east-1"
+    else S3.create_bucket(Bucket=ATHENA_BUCKET, CreateBucketConfiguration=location)
+)
+stat(bucket)
+# print(bucket)
+
+# Create Workgroup which outputs to that Bucket (if needed)
+
+lwg = ATHENA.list_work_groups()
+wgs = [wg["Name"] for wg in lwg["WorkGroups"]]
+
+if ATHENA_WORKGROUP not in wgs:
+    cwg = ATHENA.create_work_group(
+        Name=ATHENA_WORKGROUP, Description="Quilt uses this for Athena SQL Queries"
+    )
+    stat(cwg)
+
+# Configure Workgroup to use that Bucket
+uwg = ATHENA.update_work_group(
+    WorkGroup=ATHENA_WORKGROUP,
+    ConfigurationUpdates={
+        "ResultConfigurationUpdates": {
+            "OutputLocation": ATHENA_URL,
+        },
+    },
+)
+stat(uwg)
+
+# Create new GLUE Database
+
+sqe = ATHENA.start_query_execution(
+    QueryString=f"create database {ATHENA_DB}",
+    ResultConfiguration={"OutputLocation": ATHENA_URL + "/queries/"},
+)
+stat(sqe)
+```
+
+    200
+    200
+    200
+
+
+## II. Granting Access to Athena
+
+By default, Quilt runs with very conservative permissions that do not allow access to [Amazon Athena](https://docs.aws.amazon.com/athena/latest/ug/what-is.html). To enable Athena SQL queries by your Quilt users, you must:
+
+### A. Create a new Athena policy.
+
+The standard [AmazonAthenaFullAccess](https://console.aws.amazon.com/iam/home#/policies/arn:aws:iam::aws:policy/AmazonAthenaFullAccess) policy is more permissive than necessary.  For production usage, we recommend creating a policy limited to only the above Bucket:
+<!--pytest-codeblocks:cont-->
+
+
+```python
+# https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/iam.html#IAM.ServiceResource.create_policy
+# https://docs.aws.amazon.com/athena/latest/ug/workgroups-access.html
+
+STS = boto3.client("sts")
+ACCOUNT_ID = STS.get_caller_identity()["Account"]
+ARN_POLICY = f"arn:aws:iam::{ACCOUNT_ID}:policy/AthenaQuiltAccess"
+
+AthenaQuiltAccess = {
+    "Version": "2012-10-17",
+    "Statement": [
+        {"Effect": "Allow", "Action": ["athena:*"], "Resource": ["*"]},
+        {
+            "Effect": "Allow",
+            "Action": [
+                "glue:GetDatabase",
+                "glue:GetDatabases",
+                "glue:CreateTable",
+                "glue:DeleteTable",
+                "glue:UpdateTable",
+                "glue:GetTable",
+                "glue:GetTables",
+            ],
+            "Resource": ["*"],
```

Comment:
Some suggested restricting this to ONLY the Workgroup resource.
But as far as I can tell, Athena Workgroups do NOT have ARNs, so they can't participate in policies.
Any other suggestions?

Reply:
I'm not sure about glue but as for Athena, Workgroups have ARNs, see https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonathena.html#amazonathena-resources-for-iam-policies.

Reply:
Updated Policy to use Athena/Glue ARNs